prime-angular-frontend/src/app/modules/site-registration/pages/remote-user-page/remote-user-page.component.ts

```diff
@@ -96,6 +95,41 @@ export class RemoteUserPageComponent extends AbstractEnrolmentPage implements On
       this.enrolmentService.hasAllowedRemoteAccessLicences(licenceConfig);
   }
 
+  /**
+   * @description
+   * Deactivation guard handler. Reworked to use form, !formState
+   */
+  public canDeactivate(): Observable<boolean> | boolean {
+    const data = 'unsaved';
+    return (this.form.dirty && !this.allowRoutingWhenDirty)
+      ? this.dialog.open(ConfirmDialogComponent, { data })
+        .afterClosed()
+        .pipe(
+          map((confirmation: boolean) => {
+            this.handleDeactivation(confirmation);
+            return confirmation;
+          })
+        )
+      : true;
+  }
+
+  /**
+   * @description
+   * Form submission event handler. Reworked to use form, !formState
+   */
+  public onSubmit(): void {
+    this.hasAttemptedSubmission = true;
+
+    if (this.checkValidity(this.form)) {
+      this.onSubmitFormIsValid();
+      this.busy = this.performSubmission()
+        .pipe(tap((_) => this.form.markAsPristine()))
+        .subscribe(() => this.afterSubmitIsSuccessful());
+    } else {
+      this.onSubmitFormIsInvalid();
```

Comment:
`onSubmitFormIsValid` and `onSubmitFormIsInvalid` do not seem to be implemented anywhere, wonder  what they are used for?

Reply:
So i'm overwriting the functionality from the AbstractEnrolmentPage. This is because the form state in this component points to the parents form state and not the singular remote user.

Those methods are hooks that you can overwrite to add functionality, i chose to leave them in since if later someone wants to implement those hooks they don't have to remember where to add them in the handler.